04-gates/00-gates.md

```diff
@@ -2,4 +2,54 @@
 # Protocol gateways
 ## HTTP
 ## S3
+
+[NeoFS S3 gateway](https://github.com/nspcc-dev/neofs-s3-gw) provides API compatible with Amazon S3 cloud storage service.
+
+### Access Box scheme
+
+S3 gateway have to authenticate user requests staying in [AWS spec](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html).
+So we have the following scheme:
+
+![Authmate scheme](pic/authmate-scheme.png)
+
+
+1. User use the [neofs-s3-authmate](https://github.com/nspcc-dev/neofs-s3-gw/blob/master/docs/authmate.md)
+(Authmate) tool to get credentials (`access_key_id` and `secret_access_key`).
+2. Authmate forms "AccessBox" object (see the next section) and put it into NeoFS.
+3. Authmate output contains credentials (`access_key_id` and `secret_access_key`) that can be used with AWS CLI for example.
+4. User send request to NeoFS S3 Gateway using standard AWS tool.
+5. S3 Gateway get "AccessBox" from NeoFS by `access_key_id` and fetch Bearer Token from it.
+6. S3 Gateway use fetched Bearer token to send a request to NeoFS on behalf of user.
+
+#### Form Access Box object
+
+Actually, `AccessBox` is a regular object in NeoFS but properly formed. 
+It contains encrypted  Bearer token (`BT`), Session tokens (`STs`) and `secret_access_key`.
+
+The credentials are formed by following steps:
+
+1. User provides to Authmate `BT`, `STs` and `s3gw_public_key_k`(public keys of gates that will be able 
+to handle credentials) that will be used when user sends requests via S3 Gateway.
+2. Authmate:
+   1. Generates `secret_access_key` (it's 32 random bytes), and P256 (secp256r1) key pair 
```

Comment:
Generates a `secret_access_key` (it's 32 random bytes) and a P256 (secp256r1) key pair